sql/2020/07_SEO/pages_wpt_bodies_hreflang_by_device_and_http_header_value.sql

```diff
@@ -0,0 +1,52 @@
+#standardSQL
+# page wpt_bodies metrics grouped by device and hreflang value in http headers
+
+# helper to create percent fields
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+# returns all the data we need from _wpt_bodies
+CREATE TEMPORARY FUNCTION get_wpt_bodies_info(wpt_bodies_string STRING)
+RETURNS STRUCT<
+    hreflangs ARRAY<STRING>
+> LANGUAGE js AS '''
+var result = {};
+
+try {
+    var wpt_bodies = JSON.parse(wpt_bodies_string);
+
+    if (Array.isArray(wpt_bodies) || typeof wpt_bodies != 'object') return result;
+
+    if (wpt_bodies.hreflangs && wpt_bodies.hreflangs.http_header && wpt_bodies.hreflangs.http_header.values) {
+        result.hreflangs = wpt_bodies.hreflangs.http_header.values;
+    }
+
+} catch (e) {}
+return result;
+''';
+
+SELECT
+client,
+hreflang,
```

Comment:
Collapse interchangeable hreflangs
```suggestion
NORMALIZE_AND_CASEFOLD(hreflang) AS hreflang,
```